packit_service/service/api/webhooks.py

```diff
@@ -206,31 +208,86 @@ def post(self):
 
         return "Webhook accepted. We thank you, Gitlab.", HTTPStatus.ACCEPTED
 
-    @staticmethod
-    def validate_token():
+    def create_issue_token(self):
+        project_data = request.json["project"]
+
+        namespace, repo_name = project_data["path_with_namespace"].split("/")
+        http_url = project_data["git_http_url"]
+
+        project_token = ProjectTokenModel.get_project(
+            namespace=namespace, repo_name=repo_name, project_url=http_url
+        )
+
+        if not project_token:
+            token = jwt.encode(
+                {"namespace": namespace, "repo_name": repo_name},
+                "secret",
+                algorithm="HS256",
+            )
+
+            _ = ProjectTokenModel.create(
+                token=token,
+                namespace=namespace,
+                repo_name=repo_name,
+                project_url=http_url,
+            )
+
+            project = config.get_project(url=self.http_url)
+            logger.info("CONFIDENTIAL ISSUE CREATED HERE")
+            project.create_issue(
+                title="Packit-Service",
+                body=f"Set token {token} in webhook settings.",
```

Comment:
- Can we guess the link?
- Is this really clear? GitLab settings are a bit complicated and are divided into a lot of pages.
- What is the name of the variable sent with the webhook?
- I would use `  for token value or other markdown formating for better readability and being able to copy-paste it easily.

Reply:
> * Can we guess the link?

Yes, making users click the link is good idea.

> * Is this really clear? GitLab settings are a bit complicated and are divided into a lot of pages.

How about,

Title- Packit-Service Authentication

Body -

To configure Packit-Service with `repo_name` you need to configure the webhook settings.
Head to https://urltowebhook.com and add the secret token `secret_token` to authenticate
requests coming to Packit.

Grant [packit_user](https://userlink.com) `admin` permissions on the {owner}/{project} project.
You can add Packit user by clicking [here](link to the add_users settings) 
  manually to match the Packit configuration.

OR

Body-

Please following these steps to configure Packit correctly

- Packit needs to validate requests coming from `repo_name`, Head to https://urltowebhook.com 
and add the secret_token as `secret_token`, to webhook settings.
- Grant [packit_user](https://userlink.com) `admin` permissions on the {owner}/{project} project.
- You can add Packit user by clicking [here](link to the add_users settings) 
  manually to match the Packit configuration.


> * What is the name of the variable sent with the webhook?

👍

Reply:
Nice.

I like the first version of the text. It's more friendly.



> > * What is the name of the variable sent with the webhook?
> 
> it's secret_token.

Isn't UPPER_CASE more common?

Reply:
Yup! Changed the body of the issue.

Reply:
OK, I misunderstood that completely. We should use `Secret token` or `Secret token field` since it's the name of the settings field, not some key-value pair user needs to specify...

Reply:
Ah okay!

Reply:
Done!